mobile_verifier/src/reward_shares.rs

```diff
@@ -203,52 +204,147 @@ pub fn dc_to_mobile_bones(dc_amount: Decimal, mobile_bone_price: Decimal) -> Dec
+    pub async fn aggregate_points(
+        hex_streams: &impl CoveredHexStream,
+        heartbeats: impl Stream<Item = Result<HeartbeatReward, sqlx::Error>>,
+        speedtests: &SpeedtestAverages,
+        period_end: DateTime<Utc>,
+    ) -> Result<Self, sqlx::Error> {
+        let mut heartbeats = std::pin::pin!(heartbeats);
+        let mut covered_hexes = CoveredHexes::default();
+        let mut coverage_points = HashMap::new();
+        while let Some(heartbeat) = heartbeats.next().await.transpose()? {
+            let speedtest_multiplier = speedtests
+                .get_average(&heartbeat.hotspot_key)
                 .as_ref()
                 .map_or(Decimal::ZERO, SpeedtestAverage::reward_multiplier);
-            *poc_shares
-                .hotspot_shares
-                .entry(heartbeat_reward.hotspot_key.clone())
-                .or_default()
-                .radio_shares
-                .entry(heartbeat_reward.cbsd_id.clone())
-                .or_default() += heartbeat_reward.reward_weight() * speedmultiplier;
+
+            if speedtest_multiplier.is_zero() {
+                continue;
+            }
+
+            let seniority = hex_streams
+                .fetch_seniority(heartbeat.key(), period_end)
+                .await?;
+            let covered_hex_stream = hex_streams
+                .covered_hex_stream(heartbeat.key(), &heartbeat.coverage_object, &seniority)
+                .await?;
+            covered_hexes
+                .aggregate_coverage(&heartbeat.hotspot_key, covered_hex_stream)
+                .await?;
+            let opt_cbsd_id = heartbeat.key().to_owned().into_cbsd_id();
+            coverage_points
+                .entry(heartbeat.hotspot_key)
+                .or_insert_with(|| HotspotPoints::new(speedtest_multiplier))
+                .radio_points
+                .insert(
+                    opt_cbsd_id,
+                    RadioPoints::new(
+                        heartbeat.reward_weight,
+                        heartbeat.coverage_object,
+                        seniority.seniority_ts,
+                    ),
+                );
+        }
+
+        for CoverageReward {
+            radio_key,
+            points,
+            hotspot,
+        } in covered_hexes.into_coverage_rewards()
+        {
+            // Guaranteed that points contains the given hotspot.
+            coverage_points
+                .get_mut(&hotspot)
+                .unwrap()
+                .radio_points
+                .get_mut(&radio_key.into_cbsd_id())
+                .unwrap()
+                .points += points;
+        }
+
+        Ok(Self { coverage_points })
+    }
+
+    /*
+    pub fn is_valid(&self, hotspot: &PublicKeyBinary) -> bool {
+        if let Some(coverage_points) = self.coverage_points.get(hotspot) {
+            !coverage_points.total_points().is_zero()
+        } else {
+            false
         }
-        Ok(poc_shares)
+    }
+     */
+
+    /// Only used for testing
```

Comment:
```suggestion
#[cfg(test)]
```
?

Reply:
can't, used in integration tests

Reply:
was not aware the `/test` directory tests weren't compiled for the test environment. that'd just make too much sense